pandas/core/indexes/multi.py

```diff
@@ -3534,7 +3534,7 @@ def _union(self, other, sort) -> MultiIndex:
         other, result_names = self._convert_can_do_setop(other)
         if (
             any(-1 in code for code in self.codes)
-            and any(-1 in code for code in self.codes)
+            and any(-1 in code for code in other.codes)
```

Comment:
great can you see if you can find a test that fails for this w/o the change and passes with?

Reply:
Probably not since this is mostly performance relevant.

Reply:
This would explain why there is no bug associated with this mistyping. Does `panda` use also performance testing/checking too? Happy to add such test if so.

Reply:
and while we are rewriting, any reason
`self.codes.any()` is not used? generally more performant and more readable

Reply:
https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#running-the-performance-test-suite

Reply:
How could we check for -1 with any?

Reply:
oh right, this is not a null check, ok then.